cloud/amazon/ec2_vpc_nat_gateway.py

```diff
@@ -0,0 +1,356 @@
+    type: string
+    sample: "nat-0d1e3a878585988f8"
+'''
+
+import datetime
+import time
+
+try:
+    import boto3
+    import botocore
+
+    HAS_BOTO_3 = True
+except ImportError:
+    HAS_BOTO_3 = False
+from distutils.version import LooseVersion
+
+if LooseVersion(botocore.__version__) < LooseVersion("1.3.14"):
+    HAS_SUFFICIENT_BOTOCORE = False
+else:
+    HAS_SUFFICIENT_BOTOCORE = True
+
+
+def get_nat_gateway_status_list(ec2_client=None, module=None):
+    """ Check if one or more NAT gateways exist with the specified attributes and return a list of matching """
+    try:
+        subnet_id = module.params.get('subnet_id')
+        allocation_id = module.params.get('allocation_id')
+        # We only pass nat_gateway_id alone if we are ensuring it is absent
+        if module.params.get('nat_gateway_id'):
+            result = ec2_client.describe_nat_gateways(NatGatewayIds=[module.params.get('nat_gateway_id')])
+            return [{'state': result.get('NatGateways')[0].get('State'),
+                     'created': result.get('NatGateways')[0].get('CreateTime')}]
+        # Checking if VPC NAT Gateway matching attributes is present
+        elif all((subnet_id, allocation_id)):
+            result = ec2_client.describe_nat_gateways(Filter=[
+                {
+                    'Name': 'subnet-id',
+                    'Values': [subnet_id]
+                }
+            ])
+            if result.get('NatGateways'):
+                gateway_list = list()
+                for nat_gateway in result.get('NatGateways'):
+                    for nat_gateway_address in nat_gateway.get('NatGatewayAddresses'):
+                        if nat_gateway_address.get('AllocationId') == allocation_id:
+                            gateway_list.append({'state': nat_gateway.get('State'),
+                                                 'created': nat_gateway.get('CreateTime'),
+                                                 'nat_gateway_id': nat_gateway.get('NatGatewayId')})
+                return gateway_list
+        return [{'state': 'absent'}]
+    except botocore.exceptions.ClientError as ce:
+        if ce.response['Error']['Code'] == "NatGatewayNotFound":
+            return [{'state': 'absent'}]
+        else:
+            module.fail_json(msg=ce.message)
+
+
+def delete_nat_gateway(ec2_client=None, module=None, nat_gateway=None):
+    """ Delete an existing NAT gateway """
+    nat_gateway_address = nat_gateway.get('NatGatewayAddresses')[0]
+    nat_gateway_id = nat_gateway['NatGatewayId']
+    results = dict(changed=True, nat_gateway_id=nat_gateway_id,
+                   public_ip=nat_gateway_address.get('PublicIp'),
+                   private_ip=nat_gateway_address.get('PrivateIp'),
+                   allocation_id=nat_gateway_address.get('AllocationId'))
+    ec2_client.delete_nat_gateway(NatGatewayId=nat_gateway_id)
+    wait = module.params.get('wait')
+    if wait:
```

Comment:
Can i suggest using a boto3 waiter here rather than 'get_nat_gateway_status_list'?

http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.Waiter.NatGatewayAvailable